core/src/test/java/org/elasticsearch/client/transport/TransportClientIT.java

```diff
@@ -97,4 +106,47 @@ public void testThatTransportClientSettingCannotBeChanged() {
             assertThat(Client.CLIENT_TYPE_SETTING_S.get(settings), is("transport"));
         }
     }
+
+    /**
+     * test that when plugins are provided that want to register
+     * {@link NamedWriteable}, those are also made known to the
+     * {@link NamedWriteableRegistry} of the transport client
+     */
+    public void testPluginNamedWriteablesRegistered() {
```

Comment:
I think it'd be nice if this was in a unit test rather than an `ESIntegTestCase`. I think it'd work just fine in one, right? If so I think it is worth adding the extra test class so we can keep them separate.

I just know *someone* is going to come back through these tests one day looking to refactor something and wonder "does this need to be in an ESIntegTestCase?"